File: scripts/sync_arbs.py
import json
from pathlib import Path

root = Path(__file__).resolve().parents[1]
ln = root / 'lib' / 'l10n'

tr_file = ln / 'app_tr.arb'
en_file = ln / 'app_en.arb'
locales = ['app_de.arb','app_es.arb','app_fr.arb','app_ar.arb','app_zh.arb']

with tr_file.open('r', encoding='utf-8') as f:
    tr = json.load(f)
with en_file.open('r', encoding='utf-8') as f:
    en = json.load(f)

added = {}
for loc in locales:
    p = ln / loc
    if not p.exists():
        print(f"Skipping missing file {p}")
        continue
    with p.open('r', encoding='utf-8') as f:
        target = json.load(f)

    changed = False
    for key, val in tr.items():
        if key.startswith('@@'):
            # section comments: copy if missing
            if key not in target:
                target[key] = val
                changed = True
        elif key.startswith('@'):
            # metadata entries (placeholders) handled when main key missing
            if key not in target:
                target[key] = val
                changed = True
        else:
            if key not in target:
                # choose translation from English if present, else Turkish
                translation = en.get(key, val)
                target[key] = translation
                # copy metadata for this key if present in tr and missing in target
                meta_key = '@' + key
                if meta_key in tr and meta_key not in target:
                    target[meta_key] = tr[meta_key]
                changed = True
    if changed:
        # write back file
        with p.open('w', encoding='utf-8') as f:
            json.dump(target, f, ensure_ascii=False, indent=4)
        added[loc] = True
    else:
        added[loc] = False

print('Done. Updated files:')
for k,v in added.items():
    print(f"{k}: {'updated' if v else 'no changes'}")

File: scripts/sync_arbs.js
const fs = require('fs');
const path = require('path');
const root = path.resolve(__dirname, '..');
const ln = path.join(root, 'lib', 'l10n');
const trFile = path.join(ln, 'app_tr.arb');
const enFile = path.join(ln, 'app_en.arb');
const locales = ['app_de.arb', 'app_es.arb', 'app_fr.arb', 'app_ar.arb', 'app_zh.arb'];

function loadJson(p) {
    return JSON.parse(fs.readFileSync(p, 'utf8'));
}

const tr = loadJson(trFile);
const en = loadJson(enFile);

const report = {};
for (const loc of locales) {
    const p = path.join(ln, loc);
    if (!fs.existsSync(p)) {
        console.log(`Skipping missing file ${p}`);
        continue;
    }
    const target = loadJson(p);
    let changed = false;
    for (const [key, val] of Object.entries(tr)) {
        if (key.startsWith('@@')) {
            if (!(key in target)) {
                target[key] = val; changed = true;
            }
            continue;
        }
        if (key.startsWith('@')) {
            if (!(key in target)) {
                target[key] = val; changed = true;
            }
            continue;
        }
        if (!(key in target)) {
            const translation = (key in en) ? en[key] : val;
            target[key] = translation;
            const metaKey = '@' + key;
            if ((metaKey in tr) && !(metaKey in target)) target[metaKey] = tr[metaKey];
            changed = true;
        }
    }
    if (changed) {
        fs.writeFileSync(p, JSON.stringify(target, null, 4), 'utf8');
    }
    report[loc] = changed;
}
console.log('Done. Updated files:');
for (const k of Object.keys(report)) {
    console.log(`${k}: ${report[k] ? 'updated' : 'no changes'}`);
}
